alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/implementations/geometry/RangeExtension.kt

```diff
@@ -0,0 +1,98 @@
+ */
+fun <T : Comparable<T>> ClosedRange<T>.contains(other: ClosedRange<T>): Boolean =
+    start <= other.start && endInclusive >= other.endInclusive
+
+/**
+ * Checks whether the range intersects the [other] one.
+ */
+fun <T : Comparable<T>> ClosedRange<T>.intersects(other: ClosedRange<T>): Boolean =
+    start <= other.endInclusive && endInclusive >= other.start
+
+/*
+ * Finds the intersection between two ranges which are guaranteed to intersect.
+ */
+private fun <T : Comparable<T>> ClosedRange<T>.unsafeIntersection(other: ClosedRange<T>): ClosedRange<T> =
+    ObjectUtils.max(start, other.start)..ObjectUtils.min(endInclusive, other.endInclusive)
+
+/**
+ * Finds the intersection between two ranges, the resulting range may feature a single value
+ * (if the ranges only share an endpoint) or can be null, if they don't intersect at all.
+ */
+fun <T : Comparable<T>> ClosedRange<T>.intersection(other: ClosedRange<T>): ClosedRange<T>? = when {
+    intersects(other) -> unsafeIntersection(other)
+    else -> null
+}
+
+/**
+ * Checks whether two ranges intersect, excluding their bounds (i.e., excluding both
+ * [ClosedRange.start] and [ClosedRange.endInclusive]). This means false is returned in
+ * case the ranges share a single endpoint.
+ */
+fun <T : Comparable<T>> ClosedRange<T>.intersectsBoundsExcluded(other: ClosedRange<T>): Boolean =
+    start < other.endInclusive && endInclusive > other.start
+
+/**
+ * Performs a subtraction between ranges. The operation can produce an empty list (e.g. if
+ * the current range is contained in the [other] one), a list featuring a single element,
+ * or a list featuring two elements (e.g. if the current range contains the [other] one).
+ */
+infix operator fun <T : Comparable<T>> ClosedRange<T>.minus(other: ClosedRange<T>): List<ClosedRange<T>> = when {
+    other.contains(this) -> emptyList()
+    !intersects(other) -> listOf(this)
+    start >= other.start -> listOf(rangeFromUnordered(endInclusive, other.endInclusive))
+    endInclusive <= other.endInclusive -> listOf(rangeFromUnordered(start, other.start))
+    else -> listOf(rangeFromUnordered(start, other.start), rangeFromUnordered(endInclusive, other.endInclusive))
+}
+
+/**
+ * Subtracts all the given ranges from the current one. See [ClosedRange.minus].
+ */
+fun <T : Comparable<T>> ClosedRange<T>.subtractAll(others: List<ClosedRange<T>>): List<ClosedRange<T>> = when {
+    others.isEmpty() -> listOf(this)
+    else -> (this - others.first()).flatMap { it.subtractAll(others.drop(1)) }
+}
+
+/**
+ * Given a non empty list of points represented as vectors, this method finds the extreme
+ * coordinates (i.e. min and max coordinates) either on the X-axis or the Y-axis. [getXCoords]
+ * indicates which coordinates to extract, these are used to create the returned [ClosedRange].
+ */
+fun <V : Vector2D<V>> List<V>.findExtremeCoords(getXCoords: Boolean): ClosedRange<Double> {
```

Comment:
make this private, create a function for x and one for y: booleans do not map well to axes in an API

Reply:
Ok